### PR TITLE
New API - pre 0.16.0 outstanding issues

### PR DESCRIPTION
**Pull request by john-e-riordan**
Catching up on outstanding stuff that has come in over the last few months to bring 0.15.x up to current times.

- Changes
- Documentation
- Features
- Issues
- Etc

### CODE REVIEW

File: docs/api/sip.js.sessiondelegate.md [hunk 1]
```diff
@@ -16,8 +16,10 @@ export interface SessionDelegate
 
 |  Method | Description |
 |  --- | --- |
+|  [onBye(bye)](./sip.js.sessiondelegate.onbye.md) | Called upon receiving an incoming in dialog BYE request. |
 |  [onInfo(info)](./sip.js.sessiondelegate.oninfo.md) | Called upon receiving an incoming in dialog INFO request. |
 |  [onInvite(request, response, statusCode)](./sip.js.sessiondelegate.oninvite.md) | Called upon receiving an incoming in dialog INVITE request. |
+|  [onMessage(message)](./sip.js.sessiondelegate.onmessage.md) | Called upon receiving an incoming in dialog MESAGE request. |
```
Comment (sboily) [inline docs/api/sip.js.sessiondelegate.md:22]: small typo, MESSAGE instead MESAGE.

File: docs/api/sip.js.sessiondelegate.onmessage.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<!-- Do not edit this file. It is automatically generated by API Documenter. -->
+
+[Home](./index.md) &gt; [sip.js](./sip.js.md) &gt; [SessionDelegate](./sip.js.sessiondelegate.md) &gt; [onMessage](./sip.js.sessiondelegate.onmessage.md)
+
+## SessionDelegate.onMessage() method
+
+Called upon receiving an incoming in dialog MESAGE request.
```
Comment (sboily) [inline docs/api/sip.js.sessiondelegate.onmessage.md:7]: same here MESSAGE instead MESAGE

File: docs/api/sip.js.sessiondelegate.md (hunk 1 above)
Comment (john-e-riordan) [reply docs/api/sip.js.sessiondelegate.md]: Thanks - fixed.

File: docs/api/sip.js.sessiondelegate.onmessage.md (hunk 2 above)
Comment (john-e-riordan) [reply docs/api/sip.js.sessiondelegate.onmessage.md]: Thanks - fixed.

**Pull request merged**